frontend/src/page/Realm.tsx

```diff
@@ -27,8 +56,48 @@ export const Realm: React.FC<Props> = ({ match }) => {
     </React.Fragment>;
 };
 
-const DUMMY_NAMES: { [key: string]: string } = {
-    "lectures": "Lectures",
-    "math": "Department of Mathematics",
-    "algebra": "Algebra I",
+// Looks up each segment of the `/` separated `path` as realm and returns a list
+// of realm ids starting with 0 (root).
+const resolvePath = (path: string): number[] | null => {
+    const ids = [0];
+    for (const segment of path == "" ? [] : path.split("/")) {
+        const lastId = ids[ids.length - 1];
+        const next = REALMS[lastId].children.find(child => REALMS[child].path == segment);
+        if (!next) {
+            return null;
+        }
+        ids.push(next);
+    }
+
+    return ids;
+};
+
+type Realm = {
+    path: string,
+    name: string,
+    parent: number,
+    children: number[],
+};
```

Comment:
Any reason why `parent` and `children` are not (an array of) references, directly?

Reply:
Oops, Rust leaks into my JS I guess. But I'd just keep it like that for now. Will be replaced anyway as soon as we get that data from the API.